firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -121,7 +135,7 @@ private void preConditionChecks() {
    * @return a {@link FirebaseInstallationsApi} instance
    */
   @NonNull
-  public static FirebaseInstallations getInstance() {
+  public static FirebaseInstallations getInstance() throws FirebaseInstallationsException {
```

Comment:
Using an incompatible version of the Instance-ID SDK in parallel can break Firebase Targeting, as the same App-Instance will get 2 different identifiers. Ensuring that we do not break Firebase Targeting had the highest priority when discussing the design of FIS SDKs. There is a rather short summary in go/fis-iid-incompatible-versions.
In that sense, don't you think it is desirable to prevent this situation with a breaking (checked) exception?

Reply:
I guess one could argue either way, but imo the severity of the condition that causes an exception to be thrown is orthogonal to the type of exception to throw(`checked/unchecked`).

To quote effective java(go/ej3e-70):
```
 use checked exceptions for conditions from which the caller
can reasonably be expected to recover.
...By confronting the user with a checked exception, the API designer presents a
mandate to recover from the condition
```

With this in mind, it does not sound like something that can be recovered from at runtime, and the only way to recover is to rebuild the app with a different version of iid, hence my suggestion to make it unchecked. But if there are reasonable ways to recover from this state then sure, it could be checked.

Reply:
@vkryachko Accepted your suggestion and updated the code. PTAL.